theano/link/basic.py

```diff
@@ -506,10 +549,16 @@ def accept(self, fgraph, no_recycling=None, profile=None):
         self.linkers = [linker.accept(fgraph, no_recycling) for linker in self.linkers]
         return self
 
-    def pre(self, f, inputs, order, thunk_groups):
+    def pre(
+        self,
+        f: "WrapLinker",
+        inputs: Union[List[ndarray], List[Optional[float]]],
+        order: List[Apply],
+        thunk_groups: List[Tuple[Callable]],
+    ) -> None:
         pass
 
-    def make_thunk(self, **kwargs):
+    def make_thunk(self, **kwargs) -> Tuple[Callable, List[Container], List[Container]]:
```

Comment:
Not as precise as the one that was already annotated in the base class

Reply:
See the link in [this comment](https://github.com/pymc-devs/Theano-PyMC/pull/272#discussion_r564907981) for more specific type information regarding this case.

Reply:
I changed this one to `Tuple[Callable[[], NoReturn], List[Container], List[Container]]` to match the type hints from base class. Should this be defined as its own type too instead?

Reply:
Yeah, a `ThunkType` might be reasonable.

Reply:
Done with `ThunkType = Tuple[Callable[[], NoReturn], List["Container"], List["Container"]]`